### DISCUSSION TITLE
Naming convention for Table of Contents files when reporting_entity_name is same but reporting_entity_type is different

### DISCUSSION THREAD

**Discussion by srn-shekar** [CMSgov/price-transparency-guide#455] [General]
We will be creating 2 separate TOC files with multiple plans for our Self Insured (ASO) and Fully Insured plans, and have a scenario where the "reporting_entity_name" will be the same for both TOCs although "reporting_entity_type" will be different.

for example:

For fully insured:
	"reporting_entity_name": "ACME Inc",
	"reporting_entity_type": "Health Insurance Issuer",

For ASO or ASC accounts:
	"reporting_entity_name": "ACME Inc",
	"reporting_entity_type": "Third Party",
	

Based on the required naming standard for the table-of-contents file, we are looking at this scenario where both TOCs will have the same name.

2022-04-12_ACME Inc_index.json

Could you please advise how these files should be named?

**Comment by Devyandu** [comment]
@srn-shekar 
The payer name is in the TOC file name, so you would need to create separate TOC files for each one of your ASO Accounts (payers). 
All fully insured plans would be in a single TOC file.

So, if you had 8 ASO accounts then you would have a total of 9 TOC files.
The eight ASO TOCs would have their respective ASO account's names in the TOC file name, the ASO account is the risk carrying entity so they are considered a payer.
The one fully insured TOC will have Acme Inc in the payer name (if that is the risk carrying entity)

All 9 files can have the same reporting entity Acme Inc (if that is indeed the entity which is producing all 9 files).

PS: I am not with CMS, but working to implement the regulation for a plan with ASO accounts.

**Reply by srn-shekar** [reply]
Thank you @Devyandu, this is very helpful. Really appreciate the quick response.

**Reply by retjami** [reply]
@shaselton @shaselton-usds can you please weigh in on this? My understanding of CMS's intent on the introduction of the TOC is to group like rates into as few files as possible to avoid unnecessarily creating many files of redundant data. Since this is all publicly available data we see no need to create TOCs for each ASO account (our UI allows for search by EIN providing easy access to all files needed by each account without creating redundant files), however the current filename guideline is a challenge. We planned on publishing 2 TOCs with the difference being reporting entity type' Health Insurance Issuer' for our fully insured business and 'Third Party' for our ASO business. 
CMS Filename Guideline <YYYY-MM-DD>_<payer or issuer name>_index.<file extension>
would something like this be considered to allow for this?

YYYY-MM-DD_<payer or issuer name>_<reporting_entity_type>_index.<file extension>

or maybe even more appropriate to be more in line with describing each TOC file and how they differ based on schema values

YYYY-MM-DD_<reporting_entity>_<reporting_entity_type>_index.<file extension>

**Reply by RLTx1391** [reply]
@shaselton-usds Any additional guidance here? @retjami were you able to work out a solution here?

**Reply by RLTx1391** [reply]
Just wanted to  bump this again, that creating TOCs for each ASO accounts leads to duplication, which is something the schema changes have been trying to prevent.
